### PR TITLE
Create `social.trades`

### PR DESCRIPTION
**Pull request by hildobby**
Create `social.trades` (renamed from `sofi.trades` as per @andrewhong5297's feedback), a sector for all trades from SocialFi projects including friend.tech, post.tech, Star Arena, friend3, hub3 & Cipher

### CODE REVIEW

File: models/social/arbitrum/cipher_arbitrum_base_trades.sql
Comment (jeff-dude) [inline models/social/arbitrum/cipher_arbitrum_base_trades.sql:1]: can we move all models to the new directory format?
`models/_sector/social/<blockchain>/`

can then use `_schema.yml` and `_source.yml` in each blockchain directory

Comment (jeff-dude) [reply models/social/arbitrum/cipher_arbitrum_base_trades.sql:1]: will only comment once here, but applies for all models

File: models/social/arbitrum/cipher_arbitrum_base_trades.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{{ config(
+    schema = 'cipher_arbitrum',
+    tags = ['dunesql'],
```
Comment (jeff-dude) [inline models/social/arbitrum/cipher_arbitrum_base_trades.sql:3]: can remove dunesql tags (again, will only comment here to keep clean)

File: models/social/arbitrum/cipher_arbitrum_base_trades.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    alias = alias('base_trades')
+    )
```
Comment (jeff-dude) [inline models/social/arbitrum/cipher_arbitrum_base_trades.sql:5]: you're missing all the incremental config setup here (and other project level spells)

File: models/social/arbitrum/cipher_arbitrum_base_trades.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+, protocolEthAmount/1e18 AS protocol_fee_amount
+, 0x0000000000000000000000000000000000000000 AS currency_contract
+, 'ETH' AS currency_symbol
+, supply
+, evt_tx_hash AS tx_hash
+, evt_index
+, contract_address
+FROM {{source('cipher_arbitrum', 'Cipher_evt_Trade')}}
+{% if is_incremental() %}
+WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
```
Comment (jeff-dude) [inline models/social/arbitrum/cipher_arbitrum_base_trades.sql:29]: please use the new incremental filter
```
    {% if is_incremental() %}
    WHERE
        {{ incremental_predicate('<column>') }}
    {% endif %}
```

File: models/social/arbitrum/social_arbitrum_schema.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+models:
+  - name: cipher_arbitrum_base_trades
+    meta:
+      blockchain: arbitrum
+      sector: social
+      project: cipher
+      contributors: hildobby
+    config:
+      tags: ['table', 'social', 'arbitrum', 'trades', 'cipher']
+    description: "Cipher basic trades info"
```
Comment (jeff-dude) [inline models/social/arbitrum/social_arbitrum_schema.yml:12]: once you set unique keys for incremental, be sure to add test here (and all models that need it)

File: models/social/arbitrum/social_arbitrum_trades.sql
Comment (jeff-dude) [inline models/social/arbitrum/social_arbitrum_trades.sql:1]: i think this one should be the view, while upstream are incremental

File: models/social/arbitrum/social_arbitrum_trades.sql [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    )
+}}
+
+{% set base_models = [
+    (ref('cipher_arbitrum_base_trades'))
+    , (ref('post_tech_arbitrum_base_trades'))
+] %}
+
+WITH trades AS (
+    {{enrich_social_trades('arbitrum', base_models, source('arbitrum', 'transactions'))}}
```
Comment (jeff-dude) [inline models/social/arbitrum/social_arbitrum_trades.sql:19]: we shouldn't enrich quite yet, that should be saved for `social_trades` final spell

File: models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+, CAST(varbinary_to_int256(bytearray_ltrim(bytearray_substring(logs.data, 1 + 32 * 3, 32))) AS UINT256) AS share_amount
+, CAST(varbinary_to_int256(bytearray_ltrim(bytearray_substring(logs.data, 1 + 32 * 6, 32)))/1e18 AS double) AS subject_fee_amount
+, CAST(varbinary_to_int256(bytearray_ltrim(bytearray_substring(logs.data, 1 + 32 * 7, 32)))/1e18 AS double) AS protocol_fee_amount
+, 0x0000000000000000000000000000000000000000 AS currency_contract
+, CAST(varbinary_to_int256(bytearray_ltrim(bytearray_substring(logs.data, 1 + 32 * 8, 32))) AS UINT256) AS supply
+, txs.hash AS tx_hash
+, logs.index AS evt_index
+, txs.to AS contract_address
+FROM {{ source('avalanche_c', 'transactions') }} txs
+LEFT JOIN {{ source('avalanche_c', 'logs') }} logs ON logs.block_number = txs.block_number
```
Comment (jeff-dude) [inline models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql:27]: can this be inner join?

File: models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    AND logs.tx_hash = txs.hash
+    AND txs.to = 0xa481b139a1a654ca19d2074f174f17d7534e8cec
+    AND txs.success
+WHERE logs.topic0 = 0xc9d4f93ded9b42fa24561e02b2a40f720f71601eb1b3f7b3fd4eff20877639ee
```
Comment (jeff-dude) [inline models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql:31]: this filter should be in the join conditions instead

File: models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    {% if is_incremental() %}
+    AND txs.block_time >= date_trunc('day', now() - interval '7' day)
+    AND logs.block_time >= date_trunc('day', now() - interval '7' day)
```
Comment (jeff-dude) [inline models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql:34]: this filter should be in the join conditions instead

File: models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    {% else %}
+    AND txs.block_time >= TIMESTAMP '{{stars_arena_start_date}}'
+    AND logs.block_time >= TIMESTAMP '{{stars_arena_start_date}}'
```
Comment (jeff-dude) [inline models/social/avalanche_c/stars_arena_avalanche_c_base_trades.sql:37]: this filter should be in the join conditions instead

File: models/social/arbitrum/social_arbitrum_trades.sql (hunk 5 above)
Comment (jeff-dude) [reply models/social/arbitrum/social_arbitrum_trades.sql:19]: applies to all blockchains

**Review by jeff-dude — CHANGES_REQUESTED** [review]
left some feedback below, that'll apply universally

File: models/social/arbitrum/social_arbitrum_trades.sql (hunk 5 above)
Comment (hildobby) [reply models/social/arbitrum/social_arbitrum_trades.sql]: Unlike dex.trades which only uses crosschain spells to enrich, this one uses the .transactions table for every chain. So given the specific enrichment inputs for each chain anyways, doesn't it make sense to enrich at a chain level here?

**Comment by jeff-dude** [comment]
> since i've let this one sit for awhile, i've gone ahead and pushed changes to fit to latest sector structure. i didn't test locally, so could fail CI, but we can iterate from here
> 
> let me know if this is still valid for what you want

CI ran fine, if you wanted to validate output data

**Comment by hildobby** [comment]
> > since i've let this one sit for awhile, i've gone ahead and pushed changes to fit to latest sector structure. i didn't test locally, so could fail CI, but we can iterate from here
> > let me know if this is still valid for what you want
> 
> CI ran fine, if you wanted to validate output data

Yeah, looked over the results and those LGTM, imo good to merge now

**Pull request merged**